### PR TITLE
handle signals in autoupdater

### PR DESCRIPTION
**Pull request by tc80**
If the bot receives SIGINT/SIGTERM, it will not exit until it is finished processing its current package.  This will help avoid inconsistent KV states.

(this has not been tested)

### CODE REVIEW

File: cmd/autoupdate/main.go [hunk 1]
```diff
@@ -65,48 +67,58 @@ func main() {
...
-			if _, err := semver.Parse(latestVersion); err != nil {
-				util.Debugf(ctx, "ignoring invalid latest version: %s\n", latestVersion)
-			} else {
-				destpckg, err := packages.ReadPackageJSON(ctx, path.Join(cdnjsPath, "ajax", "libs", pckg.Name, "package.json"))
-				util.Check(err)
-				if destpckg.Version == nil || *destpckg.Version != latestVersion {
-					commitPackageVersion(ctx, pckg, latestVersion, f)
-				}
-			}
+			packages.GitPush(defaultCtx, cdnjsPath)
```
Comment (xtuc) [inline cmd/autoupdate/main.go:119]: maybe we could push after each commit too

Comment (tc80) [reply cmd/autoupdate/main.go]: all set, also cleaned up the `select` statement

File: cmd/autoupdate/main.go [hunk 2]
```diff
@@ -87,9 +100,15 @@ func main() {
 			}
 		}
 
+		if pckg.Autoupdate.Source == "git" {
+			util.Debugf(ctx, "running git update")
+			newVersionsToCommit, latestVersion = updateGit(ctx, pckg)
+		}
```
Comment (xtuc) [inline cmd/autoupdate/main.go:106]: is this duplicated now?

Comment (tc80) [reply cmd/autoupdate/main.go]: thanks, good call. Didn't notice when I merged

**Pull request merged**